runtime/src/main/kotlin/kotlin/native/internal/KFunctionImpl.kt

```diff
@@ -7,19 +7,28 @@ package kotlin.native.internal
 
 import kotlin.reflect.KFunction
 import kotlin.reflect.KType
+import kotlin.reflect.KClass
 
 @FixmeReflection
-open class KFunctionImpl<out R>(override val name: String, val fqName: String, val bound: Boolean, val receiver: Any?,
-                                override val returnType: KType): KFunction<R> {
+internal abstract class KFunctionImpl<out R>(
+        override val name: String, val fqName: String, val receiver: Any?,
+        val arity: Int, val flags: Int, override val returnType: KType
+): KFunction<R> {
     override fun equals(other: Any?): Boolean {
         if (other !is KFunctionImpl<*>) return false
-        return fqName == other.fqName && bound == other.bound && receiver == other.receiver
+        return fqName == other.fqName && receiver == other.receiver
+                && arity == other.arity && flags == other.flags
     }
 
-    override fun hashCode(): Int {
-        return (fqName.hashCode() * 31 + if (bound) 1 else 0) * 31 + receiver.hashCode()
+    private fun horner(x: Int, vararg coeffs: Int): Int {
```

Comment:
The name is kinda confusing.

Reply:
Sure, I'll rename to a more general `evaluatePolynom`.